src/components/IterableEditComponent.tsx

```diff
@@ -524,12 +522,27 @@ export class IterableEditComponent extends React.Component<ImageUploadProps | It
 
     addAtIndex = (index: any) => {
         const clone = _.cloneDeep(this.state.model)
-        if (this.props.field.iterabletype.type === "nested") {
-            clone.splice(index, 0, {});
+        clone.splice(index, 0, this.getIterableDefaultValue(this.props.field.iterabletype))
+        this.props.modelChanged(clone)
+    }
+
```

Comment:
This method will return corresponding default value, based on iterableType. Object for nested type. string or boolean for others